cmd/exec.go

```diff
@@ -0,0 +1,196 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/open-policy-agent/opa/cmd/internal/exec"
+	"github.com/open-policy-agent/opa/internal/config"
+	internal_logging "github.com/open-policy-agent/opa/internal/logging"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/plugins/bundle"
+	"github.com/open-policy-agent/opa/plugins/discovery"
+	"github.com/open-policy-agent/opa/plugins/logs"
+	"github.com/open-policy-agent/opa/plugins/status"
+	"github.com/open-policy-agent/opa/sdk"
+	"github.com/open-policy-agent/opa/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+
+	params := &exec.Params{
+		Output:       os.Stdout,
+		OutputFormat: util.NewEnumFlag("pretty", []string{"pretty", "json"}),
+		LogLevel:     util.NewEnumFlag("error", []string{"debug", "info", "error"}),
+		LogFormat:    util.NewEnumFlag("json", []string{"text", "json", "json-pretty"}),
+	}
+
+	var cmd = &cobra.Command{
+		Use:   `exec <path> [<path> [...]]`,
+		Short: "Execute against input files",
+		Long: `Execute against input files.
+
+The 'exec' command executes OPA against one or more input files. If the paths
+refer to directories, OPA will execute contains files contained inside those
+directories, recursively.
+
+The 'exec' command accepts a --config-file/-c or series of --set options as
+arguments. These options behave the same as way as 'opa run'. Since the 'exec'
+command is intended to execute OPA in a one-shot mode, the 'exec' command will
+manually trigger plugins before and after policy execution:
+
+* Discovery (before)
+* Bundle (before)
+* Status (before)
+* Decision Logs (after)
+
+By default, the 'exec' command executes the "default decision" (specified in
+the OPA configuration) against each input file.`,
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Viper has some extra field for validation args, `Args:`:

We could put that code there, or use a generic `MinimumNArgs(1)` there; I'm assuming the latter will not give us such a nice error message, though.

👉 https://github.com/spf13/cobra/blob/master/user_guide.md#positional-and-custom-arguments